src/components/10-atoms/input-text/index.js

```diff
@@ -165,13 +251,14 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
             value="${value}"
             placeholder="${placeholder}"
             aria-required="${required}"
+            maxlength="${maxLength}"
             ?disabled="${disabled}"
             @input="${this.handleInput}"
             @focus="${this.handleFocus}"
             @blur="${this.handleBlur}"
           />
           ${
-            checkMark
+            checkMark && this.isNotAtMaxLength()
```

Comment:
I suggest to wrap this composite condition in a single helper method `showCheckmark` and use that method both here and in line 228